packages/salesforcedx-vscode-core/src/commands/forceGenerateFauxClasses.ts

```diff
@@ -61,18 +119,40 @@ class ForceGenerateFauxClassesExecutor extends SfdxCommandletExecutor<{}> {
     }
     ForceGenerateFauxClassesExecutor.isActive = false;
     this.logMetric(execution.command.logName);
+
     return;
   }
 }
 
 const workspaceChecker = new SfdxWorkspaceChecker();
 
-export async function forceGenerateFauxClassesCreate(explorerDir?: any) {
-  const parameterGatherer = new EmptyParametersGatherer();
+export async function forceGenerateFauxClassesCreate(
+  source?: SObjectRefreshSource
+) {
+  const parameterGatherer = new SObjectRefreshGatherer(source);
   const commandlet = new SfdxCommandlet(
     workspaceChecker,
     parameterGatherer,
     new ForceGenerateFauxClassesExecutor()
   );
   await commandlet.run();
 }
+
+export async function initSObjectDefinitions(projectPath: string) {
+  const hasDefaultUsernameSet =
+    (await getDefaultUsernameOrAlias()) !== undefined;
+  if (projectPath && hasDefaultUsernameSet) {
+    const sobjectFolder = path.join(
+      projectPath,
+      SFDX_DIR,
+      TOOLS_DIR,
+      SOBJECTS_DIR
+    );
+    if (!fs.existsSync(sobjectFolder)) {
+      vscode.commands.executeCommand(
```

Comment:
In the execute method of the command there's stuff like handling the channel output, prevention of running a refresh while one is already active, the notification service, task view service and showing progress that I figured could be leveraged rather than duplicated. Is there a reason to not do it this way?